src/main/java/mekanism/common/content/gear/mekatool/ModuleVeinMiningUnit.java

```diff
@@ -49,30 +51,40 @@ public int getExcavationRange() {
         return excavationRange.get().getRange();
     }
 
-    public static Set<BlockPos> findPositions(BlockState state, BlockPos location, Level world, int maxRange) {
+    public static Set<BlockPos> findPositions(Level world, Set<BlockPos> initial, boolean extended, int maxRange) {
         Set<BlockPos> found = new LinkedHashSet<>();
         Set<BlockPos> openSet = new LinkedHashSet<>();
-        openSet.add(location);
-        Block startBlock = state.getBlock();
-        int maxCount = MekanismConfig.gear.disassemblerMiningCount.get() - 1;
+        openSet.addAll(initial);
+        int maxCount = (MekanismConfig.gear.disassemblerMiningCount.get() - 1) * initial.size();
+        AtomicInteger next = new AtomicInteger(openSet.size());
+        AtomicInteger dist = new AtomicInteger();
```

Comment:
For clarification of this change am I correct in understanding that this is mainly to avoid having to recalculate the distance from the current block being done? Or is this for purposes of making it so each blasting one has their own distances calculated?

This also seems to have some minor side effects as for calculating distance previously we used a double whereas this is clamping it to an int which I feel like might have some small differences near the edge of larger spheres.

Reply:
So this is what I mentioned in the description as the biggest change to existing functionality. Basically the vein mining is doing a breadth first traversal, all of the initial positions are distance 0, all of the positions you can get to from those are distance 1, and so on. I started using minimum path distance instead of euclidean distance because I wasn't sure which point to calculate from. You could do closest matching block in the initial set, but that would be expensive and with path distance so easily available I switched to that. But I actually like it better as it limits lightning chains to the max distance (for extended mode, doesn't really affect ore veins) which makes sense to me. 
Another thing I forgot to mention is when doing it with Max options it lagged a second or two (my pc is a potato) but I think that had more to do with spawning 8k lightning particles and not breaking that many blocks, so might be a good idea to still limit lightning effects to the original max instead of the increases max.

Reply:
Also should mention that they're only AtomicInteger's to get rid of the stupid java scope warnings, I'm pretty sure it works fine as regular ints since unless I completely misunderstand how java works this is still single-threaded. For performance I should probably change this back.

Reply:
Another issue with this is that the atomic disassembler does not use common logic and will still be doing vein mining the old way, I think I'm going to see how easy it is to convert to use this logic.

Reply:
I think I am fine (though going to think about this further) assuming I am understanding correctly that path length is effectively "how many boxes outside" it is and not that to go one box outside it counts as two given it has to go over one and then up one (which for larger sizes will end up scaling a lot quicker).

Reply:
The obsidian with torches are path length 7 apart.

![Minecraft_ 1 18 2 - Singleplayer 5_9_2022 4_10_56 PM](https://user-images.githubusercontent.com/105073512/167499680-148d4f9f-0118-4377-8889-2240d1f88eb9.png)

Reply:
Does that basically mean it is a count of how many blocks have been mined so far not counting the source block?

Reply:
The latter, and if there was an obsidian where the cursor is it would be length 3

Reply:
https://en.m.wikipedia.org/wiki/Shortest_path_problem
Breadth first traversal is basically special case djikstra's with assumed path weight of 1, we could switch to using a higher distance for diagonals, but that would require a priority queue instead of regular queue (openSet). I don't think that really adds anything except that it would look more like a sphere when it's all one material and it doesn't hit the max. Checking the euclidean distance would still require having a point to calculate it from. Which area mining makes much more difficult, but we could still do it from the target block or centroid. But I honestly like it being path length along the vein more.

Reply:
Yeah I am aware of the shortest path problem I just forgot about the case of vein mining where potentially things might loop like that instead of just being a giant orb that mines as much as possible (and wanted to clarify I was understanding correctly with that edit). But yeah I think path length makes more sense than Euclidean distance for the cost of how vein mining works as it is more accurate for the actual "distance"